widget/card.go

```diff
@@ -0,0 +1,204 @@
+package widget
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/theme"
+)
+
+type cardRenderer struct {
+	*widget.ShadowingRenderer
+
+	header, subHeader *canvas.Text
+
+	card *CardContainer
+}
+
+const (
+	cardMediaHeight = 128
+)
+
+// MinSize calculates the minimum size of a card.
+// This is based on the contained text, image and content.
+func (c *cardRenderer) MinSize() fyne.Size {
+	hasHeader := c.card.Title != ""
+	hasSubHeader := c.card.SubTitle != ""
+	hasImage := c.card.Image != nil
+	hasContent := c.card.Content != nil
+
+	if !hasHeader && !hasSubHeader && !hasContent {
+		return fyne.NewSize(c.card.Image.MinSize().Width+theme.Padding(), cardMediaHeight+theme.Padding())
```

Comment:
I think this will cause a segfault if `c.card.Image` is nil, since you don't check that in the conditional.

Reply:
Hmm, fair point - I did not think of the case when someone would want a card with no content